Adding X-RELEVANT-ROLES header

dmnjohns opened the pull request:
This PR adds support for relevant roles. Relevant roles are roles which grant a request access to a resource. Technically, these roles are the intersection of roles which belong to the user (i.e., the values of the X-Roles header) and roles which are required to access a resource (i.e., the values of rax:roles). Relevant roles will be added to the X-Relevant-Roles request header.

dmnjohns commented:
The failing tests appear to be tests that verify the path through the state machine. My change added another step in the path. What would you like to do with those tests? For the moment, I might just add the state to the assertions to get the tests to pass.

RackerWilliams commented:
@dmnjohns  : Please fix the tests by adding the correct assertions.

RackerWilliams commented on RELEASE.md:
```diff
@@ -3,6 +3,9 @@
    level to allow capturing a header with the result of the XPath. The XPaths can simultaneously introspect headers, uri, methods, and body content (xml or json).
+1. Added support for `X-Relevant-Roles`, a header which will be added to the request any time `rax:roles` are being used.
+   `X-Relevant-Roles` are the values of the `X-Roles` header which match a `rax:roles` value for the resource being requested.
+   In other words, `X-Relevant-Roles` are the user roles which granted access to the resource.
```
You should mention that capture head should be enabled for relevant roles to work.

dmnjohns commented:
I believe I have addressed all of the code review issues.

While fixing the tests, I actually noticed that the XSLT code I had written did not order the sequence set used in the XPath expression it generated, which led to more states being created than were necessary. So I am now ordering the sequence set.

As for the `TestConfig`, it appears that `enableCaptureHeaderExtension` defaults to true anyway, so setting it explicitly on the configs in `RaxRolesBehaviors` should not change any behavior, but it does make things more explicit. So that's good.

RackerWilliams commented on core/src/main/resources/xsl/raxRoles.xsl:
```diff
@@ -113,15 +113,26 @@
+            <xsl:when test="not('#all' = $roles)">
+                <xsl:variable name="escaped" as="xsd:string*" select="for $r in $roles return replace($r, '&apos;&apos;', '&apos;&apos;&apos;&apos;')"/>
+                <xsl:variable name="quoted" as="xsd:string*" select="for $r in $escaped return concat('&apos;&apos;', rax:transformNBSP($r), '&apos;&apos;')"/>
+                <xsl:variable name="ordered" as="xsd:string*"><xsl:perform-sort select="$quoted"><xsl:sort/></xsl:perform-sort></xsl:variable>
```
Nice touch on putting these in order.
